deepchecks/vision/vision_data/utils.py

```diff
@@ -0,0 +1,173 @@
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Utils module for VisionData functionalities."""
+import sys
+import typing as t
+from collections import Counter
+from enum import Enum
+from numbers import Number
+
+import numpy as np
+from typing_extensions import TypedDict
+
+from deepchecks.core.errors import DatasetValidationError
+
+
+class TaskType(Enum):
+    """Enum containing supported task types."""
+
+    CLASSIFICATION = 'classification'
+    OBJECT_DETECTION = 'object_detection'
+    SEMANTIC_SEGMENTATION = 'semantic_segmentation'
+    OTHER = 'other'
+
+    @classmethod
+    def values(cls):
+        """Return all values of the enum."""
+        return [e.value for e in TaskType]
+
+
+class BatchOutputFormat(TypedDict):
+    """Batch output format required by deepchecks."""
+
+    images: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    labels: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    predictions: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    additional_data: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    embeddings: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    image_identifiers: t.Optional[t.Union[np.ndarray, t.Sequence]]
+
+
+def sequence_to_numpy(data: t.Optional[t.Sequence], expected_dtype=None, expected_ndim_per_object=None) -> \
+        t.Optional[t.List]:
+    """Convert a sequence containing some type of array to a List of numpy arrays.
+
+    Returns
+    -------
+    t.Optional[t.Sequence]
+        A list of numpy arrays of the provided data.
+    """
+    if data is None:
+        return None
+    return [object_to_numpy(x, expected_dtype, expected_ndim_per_object) for x in data]
```

Comment:
why put the objects to a list?
like I get it is because for ` x in data` but I think it is better done recursively.
for a single torch object you will run on each value instead of once doing .numpy() 
then we put those to a generic list instead of keeping it as numpy which can be helpful in some cases